(Unrelated) failures:

```
2020-08-18T02:35:07.1287641Z [ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.188 s <<< FAILURE! - in io.quarkus.devtools.commands.RemoveGradleExtensionsTest
2020-08-18T02:35:07.1295322Z [ERROR] io.quarkus.devtools.commands.RemoveGradleExtensionsTest.testPartialMatches  Time elapsed: 0.043 s  <<< FAILURE!
2020-08-18T02:35:07.1299542Z org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
2020-08-18T02:35:07.1304138Z 	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
2020-08-18T02:35:07.1308194Z 	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
2020-08-18T02:35:07.1312447Z 	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:35)
2020-08-18T02:35:07.1316982Z 	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:162)
2020-08-18T02:35:07.1327194Z 	at io.quarkus.devtools.commands.AbstractRemoveExtensionsTest.hasDependency(AbstractRemoveExtensionsTest.java:97)
2020-08-18T02:35:07.1335380Z 	at io.quarkus.devtools.commands.AbstractRemoveExtensionsTest.hasDependency(AbstractRemoveExtensionsTest.java:93)
2020-08-18T02:35:07.1342186Z 	at io.quarkus.devtools.commands.AbstractRemoveExtensionsTest.testPartialMatches(AbstractRemoveExtensionsTest.java:52)
2020-08-18T02:35:07.1347387Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-08-18T02:35:07.1385449Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-08-18T02:35:07.1385854Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-08-18T02:35:07.1386187Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-08-18T02:35:07.1386668Z 	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:686)
2020-08-18T02:35:07.1387065Z 	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
2020-08-18T02:35:07.1387410Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
2020-08-18T02:35:07.1387748Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
2020-08-18T02:35:07.1388080Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
2020-08-18T02:35:07.1388569Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
2020-08-18T02:35:07.1388901Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
2020-08-18T02:35:07.1389418Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
2020-08-18T02:35:07.1390027Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2020-08-18T02:35:07.1390379Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
2020-08-18T02:35:07.1390718Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
2020-08-18T02:35:07.1391055Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
2020-08-18T02:35:07.1391389Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
2020-08-18T02:35:07.1391713Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
2020-08-18T02:35:07.1392058Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:212)
2020-08-18T02:35:07.1392405Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1392746Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:208)
2020-08-18T02:35:07.1393084Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
2020-08-18T02:35:07.1393600Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
2020-08-18T02:35:07.1393953Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
2020-08-18T02:35:07.1394293Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1394811Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-08-18T02:35:07.1395937Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-08-18T02:35:07.1398070Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-08-18T02:35:07.1399675Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1401236Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-08-18T02:35:07.1485043Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-08-18T02:35:07.1509090Z 	at java.util.ArrayList.forEach(ArrayList.java:1259)
2020-08-18T02:35:07.1517497Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-08-18T02:35:07.1523864Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-08-18T02:35:07.1529946Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1536351Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-08-18T02:35:07.1541852Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-08-18T02:35:07.1551591Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-08-18T02:35:07.1587371Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1642789Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-08-18T02:35:07.1671118Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-08-18T02:35:07.1675330Z 	at java.util.ArrayList.forEach(ArrayList.java:1259)
2020-08-18T02:35:07.1683849Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-08-18T02:35:07.1690541Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-08-18T02:35:07.1696735Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1703914Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-08-18T02:35:07.1713704Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-08-18T02:35:07.1724751Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-08-18T02:35:07.1731763Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-08-18T02:35:07.1768951Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-08-18T02:35:07.1775299Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-08-18T02:35:07.1783438Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-08-18T02:35:07.1793344Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-08-18T02:35:07.1810154Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-08-18T02:35:07.1816371Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
2020-08-18T02:35:07.1823981Z 	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
2020-08-18T02:35:07.1838424Z 	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
2020-08-18T02:35:07.1843549Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
2020-08-18T02:35:07.1848928Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
2020-08-18T02:35:07.1859084Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
2020-08-18T02:35:07.1867929Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
2020-08-18T02:35:07.1881240Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
2020-08-18T02:35:07.1889334Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
2020-08-18T02:35:07.1902668Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2020-08-18T02:35:07.1907780Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
2020-08-18T02:35:07.1914840Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
2020-08-18T02:35:07.1916069Z 
```

--- a/extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/UndertowBuildStep.java
+++ b/extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/UndertowBuildStep.java
@@ -417,14 +417,29 @@ public class UndertowBuildStep {
         }
         if (webMetaData.getFilterMappings() != null) {
             for (FilterMappingMetaData mapping : webMetaData.getFilterMappings()) {
-                for (String m : mapping.getUrlPatterns()) {
-                    if (mapping.getDispatchers() == null || mapping.getDispatchers().isEmpty()) {
-                        recorder.addFilterURLMapping(deployment, mapping.getFilterName(), m, REQUEST);
-                    } else {
-
-                        for (DispatcherType dispatcher : mapping.getDispatchers()) {
-                            recorder.addFilterURLMapping(deployment, mapping.getFilterName(), m,
-                                    javax.servlet.DispatcherType.valueOf(dispatcher.name()));
+                List<String> urlPatterns = mapping.getUrlPatterns();
+                List<String> servletNames = mapping.getServletNames();
+                if (urlPatterns != null && !urlPatterns.isEmpty()) {
+                    for (String m : urlPatterns) {
+                        if (mapping.getDispatchers() == null || mapping.getDispatchers().isEmpty()) {
+                            recorder.addFilterURLMapping(deployment, mapping.getFilterName(), m, REQUEST);
+                        } else {
+                            for (DispatcherType dispatcher : mapping.getDispatchers()) {
+                                recorder.addFilterURLMapping(deployment, mapping.getFilterName(), m,
+                                        javax.servlet.DispatcherType.valueOf(dispatcher.name()));
+                            }
+                        }
+                    }
+                } else if (servletNames != null && !servletNames.isEmpty()) {
+                    // No URL Patterns found, map to servlet name instead
+                    for (String name : servletNames) {
+                        if (mapping.getDispatchers() == null || mapping.getDispatchers().isEmpty()) {
+                            recorder.addFilterServletNameMapping(deployment, mapping.getFilterName(), name, REQUEST);
+                        } else {
+                            for (DispatcherType dispatcher : mapping.getDispatchers()) {
+                                recorder.addFilterServletNameMapping(deployment, mapping.getFilterName(), name,
+                                        javax.servlet.DispatcherType.valueOf(dispatcher.name()));
+                            }
                         }
                     }
                 }
@@ -991,14 +1006,17 @@ public class UndertowBuildStep {
         if (description.length() > 0 || displayName.length() > 0 || smallIcon.length() > 0 || largeIcon.length() > 0) {
             dg = new DescriptionGroupMetaData();
             Descriptions descriptions = getDescription(description);
-            if (descriptions != null)
+            if (descriptions != null) {
                 dg.setDescriptions(descriptions);
+            }
             DisplayNames displayNames = getDisplayName(displayName);
-            if (displayNames != null)
+            if (displayNames != null) {
                 dg.setDisplayNames(displayNames);
+            }
             Icons icons = getIcons(smallIcon, largeIcon);
-            if (icons != null)
+            if (icons != null) {
                 dg.setIcons(icons);
+            }
         }
         return dg;
     }

--- a/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/MultipartConfigTestCase.java
+++ b/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/MultipartConfigTestCase.java
@@ -5,6 +5,7 @@ import static org.hamcrest.Matchers.is;
 
 import java.io.IOException;
 import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
 
 import javax.servlet.MultipartConfigElement;
 import javax.servlet.ServletException;
@@ -15,7 +16,6 @@ import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.Part;
 
-import org.apache.commons.codec.Charsets;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.junit.jupiter.api.extension.RegisterExtension;
@@ -48,7 +48,7 @@ public class MultipartConfigTestCase {
     @ParameterizedTest
     @ValueSource(strings = { "/foo", "/servlet-item" })
     public void testMultipartConfig(String path) {
-        given().multiPart("file", "random.txt", "Some random file".getBytes(Charsets.UTF_8))
+        given().multiPart("file", "random.txt", "Some random file".getBytes(StandardCharsets.UTF_8))
                 .when().post(path).then()
                 .statusCode(201)
                 .body(is("OK"));

--- a/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/NamedServlet.java
+++ b/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/NamedServlet.java
@@ -1,0 +1,19 @@
+package io.quarkus.undertow.test;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "named", urlPatterns = "/named")
+public class NamedServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        resp.getWriter().write("Hello World");
+    }
+
+}

--- a/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/NoUrlPatternWebFilter.java
+++ b/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/NoUrlPatternWebFilter.java
@@ -1,0 +1,20 @@
+package io.quarkus.undertow.test;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+
+@WebFilter(filterName = "MyFilter", servletNames = "named", asyncSupported = true)
+public class NoUrlPatternWebFilter implements Filter {
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        response.getWriter().write("Goodbye");
+    }
+}

--- a/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/WebFilterWithNoUrlPatternTestCase.java
+++ b/extensions/undertow/deployment/src/test/java/io/quarkus/undertow/test/WebFilterWithNoUrlPatternTestCase.java
@@ -1,0 +1,26 @@
+package io.quarkus.undertow.test;
+
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.Matchers.is;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class WebFilterWithNoUrlPatternTestCase {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(NoUrlPatternWebFilter.class, NamedServlet.class));
+
+    @Test
+    public void testFilterWithNoUrlPattern() {
+        when().get("/named").then()
+                .statusCode(200)
+                .body(is("Goodbye"));
+    }
+}

